gms/gossiper.cc

```diff
@@ -2327,6 +2388,9 @@ future<> gossiper::add_local_application_state(std::list<std::pair<application_s
                 // Add to local application state
                 local_state.add_application_state(state, value);
             }
+            local_state.add_application_state(application_state::HOST_ID, versioned_value::host_id(node.host_id));
+            auto rpc_addr = utils::fb_utilities::get_broadcast_rpc_address();
+            local_state.add_application_state(application_state::RPC_ADDRESS, versioned_value::rpcaddress(rpc_addr));
```

Comment:
Why do we need this?

This will change the two states any time a local state is changed (e.g. CACHE_HITRATES).

It's enough to add these when we add local state for the first time, which I thought we already do.